### DISCUSSION TITLE
Feature Request: Make the OSV Vulnerability Severity accessible to templating engine

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#21937] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Given `osvVulnerabilityAlerts` are enabled, I'd like to be able to use the [Vulnerability Severity](https://github.com/renovatebot/renovate/blob/1c94d81bc8e8fc0fc82da0d011e943c2672005fb/lib/workers/repository/process/vulnerabilities.ts#LL529C20-L529C20) as part of any template-eligable fields, such as `commitMessageSuffix`

This will help provide additional context to the urgency of each `[SECURITY]` PR raised when this experimental feature is enabled

**Comment by setchy** [comment]
Draft PR raised - https://github.com/renovatebot/renovate/pull/21939

**Comment by Churro** [comment]
The challenge with this idea is that one security PR often addresses **multiple** vulnerabilities. Each of them has a different severity and is added as a package rule, whereas all of them are sorted by fix version in ascending order. Since latter rules always take precedence in renovate, the rule with the highest fix version comes last. In practice, the last rule isn't necessarily the one with highest severity and showing the severity of _any_ vulnerability in the PR would be somewhat pointless.

For having the severity in the title (or analogously shown as label), it may be necessary to find a different approach than via package rules. A simple solution would be to store the highest severity together with the last package rule but that seems hacky in my eyes.

**Reply by rarkins** [reply]
We have some existing cases where we have "context aware" merging of updates into a branch. For example with `automerge`, we don't set it to `true` if the first or last upgrade is `true`, instead we set it to `true` for the branch only if _all_ are true: https://github.com/renovatebot/renovate/blob/8bdae527722375aaa865def18633a950e39dc55c/lib/workers/repository/updates/generate.ts#L366

**Reply by setchy** [reply]
Thanks @Churro for pointing this out - very helpful.

@rarkins - thanks!

I've pushed an update to the draft pr #21939 which calculates and sets the highest severity on each package rule (ie: the hacky route). I did try a few alternate approaches locally but couldn't quite find the right place to calculate this.  Between the branchify // flatten // generate logic, i couldn't find the right spot to preserve the highest severity calculation.

**Reply by setchy** [reply]
Spent some more time this morning and cleaned up the impl a lot.
